xamarin/Xamarin.Forms
# [Android/iOS] TabIndex sets focus order for TalkBack traversal

### Description of Change ###

Setting the `TabIndex` of a `VisualElement` now affects the order in which TalkBack/VoiceOver will read the accessible elements on the screen.

### Issues Resolved ### 

- fixes #2789

### API Changes ###
 
### Namespace Xamarin.Forms

#### Type Changed: Xamarin.Forms.TabIndexExtensions

Added methods:

```csharp
public static VisualElement GetFirstElementByTabIndex (System.Collections.Generic.IDictionary<System.Int32,System.Collections.Generic.List<VisualElement>> tabIndexes);
public static VisualElement GetFirstNonLayoutTabStop (System.Collections.Generic.IDictionary<System.Int32,System.Collections.Generic.List<VisualElement>> tabIndexes);
```



### Namespace Xamarin.Forms.Internals

#### Removed Type Xamarin.Forms.Internals.PageExtensions

### Namespace Xamarin.Forms.Platform.Android

#### Type Changed: Xamarin.Forms.Platform.Android.PageRenderer

Added method:

```csharp
protected override void OnLayout (bool changed, int l, int t, int r, int b);
```



### Namespace Xamarin.Forms.Platform.iOS

#### New Type: Xamarin.Forms.Platform.iOS.PageContainer

```csharp
public class PageContainer : UIKit.UIView, ObjCRuntime.INativeObject, System.IDisposable, UIKit.IUIAccessibilityContainer {
  // constructors
  public PageContainer ();
  public PageContainer (AccessibleUIViewController parent);
  // methods
  public void ClearAccessibilityElements ();
  protected override void Dispose (bool disposing);
}
```


### Breaking Changes

### Namespace Xamarin.Forms.Internals

#### Removed Type Xamarin.Forms.Internals.PageExtensions

### Platforms Affected ### 

- Core/XAML (all platforms)
- Android
- iOS

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###
1. Turn on TalkBack on your Android device or VoiceOver on your iOS device.
2. Open Control Gallery.
3. Navigate to Accessibility TabIndex test page.
4. Swipe right to traverse through the elements. Each element is labeled with its TabIndex (first) and its declaration order (second). You should see it select elements in order of TabIndex first and then DeclarationOrder for all elements with a conflicting TabIndex.

### PR Checklist ###
